src/table/use-column-widths.tsx

```diff
@@ -80,6 +82,36 @@ export function ColumnWidthsProvider({ visibleColumns, resizableColumns, childre
     }
   };
 
+  // Update getContainer when deps change to trigger new observation.
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const getContainer = useCallback(() => containerRef.current, [resizableColumns, visibleColumns, columnWidths]);
+
+  // Measure wrapper width to determine if the last column can stretch.
+  // When condition changes, updates the last column width style imperatively to avoid having a frame with improper style.
+  const [containerWidth, setContainerWidth] = useState(0);
+  useResizeObserver(getContainer, entry => {
+    if (!resizableColumns) {
+      return;
+    }
+
+    setContainerWidth(entry.contentBoxWidth);
+
+    const lastColumnId = visibleColumns[visibleColumns.length - 1]?.id;
+    const lastCellElement = lastColumnId ? cellsRef.current[lastColumnId] : null;
+    if (!lastCellElement) {
+      return;
+    }
+
+    const totalWidth = visibleColumns.reduce((total, { id }) => total + (columnWidths[id] || DEFAULT_COLUMN_WIDTH), 0);
+    const stretchLastColumn = Boolean(entry.contentBoxWidth > totalWidth);
+
+    if (stretchLastColumn && lastCellElement.style.width !== 'auto') {
+      lastCellElement.style.width = 'auto';
```

Comment:
If we moved this code here, it probably should be removed in the place where it was before. So we will not have race conditions to fragments of code doing the same.

Reply:
Ideally, yes. Practically - that is very tricky for two reasons:

1. When the value e.g. a style or a class name is set imperatively, it can be then overridden by React for the same node the next render. That's why, for example, in sticky-columns implementation we return wrapper and cell styles as properties even though those are set by the util.
2. If not setting the value with React at all, we'd have to follow the component's lifecycle. In that case the style should be set not only in the observer's callback, but also upon component's mounting and any relevant properties change.

Reply:
You have two choices

1. Add `setStretchLastColumn()` state here and let React do its job
2. Extract `applyColumnWidths()` as a separate function and call it in both resize observer and in React lifecycle

Both options sound good for me, but it needs to be only one

Reply:
I like the second approach and will explore it 👍 

How the setStretchLastColumn approach is supposed to work? I can't see how it can fix the issue.

Reply:
> How the setStretchLastColumn approach is supposed to work?

So the `stretchLastColumn`  becomes the state.

Then all code uses this state. React takes care of re-rendering

Reply:
It does re-render everything already but too late. I thought about the second suggestion and the question is how to apply the function within the React lifecycle exactly. The easiest solution I found was to execute it when the ref is assigned as it gives access to the node and fires after the thead render. This actually allows keeping all code within the util and is less complex than I originally anticipated. I will share my second revision soon.